wp-seo-main.php

```diff
@@ -27,6 +27,69 @@
 	define( 'WPSEO_CSSJS_SUFFIX', ( ( defined( 'SCRIPT_DEBUG' ) && true === SCRIPT_DEBUG ) ? '' : '.min' ) );
 }
 
+global $pagenow;
+if ( is_admin() && $pagenow === 'plugins.php' ) {
+	add_action( 'admin_init', 'yoast_seo_toggle_register_deactivate_listener' );
+}
+
+/**
+ * Listener for Yoast SEO to toggle Premium or Free when one of them is active.
+ */
+function yoast_seo_toggle_register_deactivate_listener() {
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'plugin' ) ) && ! empty( $listener_plugin ) ) {
+		$disable_plugin = '';
+
+		switch( $listener_plugin ){
+			case 'wordpress-seo/wp-seo.php':
+				$disable_plugin = 'free';
+				break;
+			case 'wordpress-seo-premium/wp-seo-premium.php':
+				$disable_plugin = 'premium';
+				break;
+		}
+
+		wp_redirect( admin_url( 'plugins.php?yoast_seo_disable=' . $disable_plugin . '&_yoast_disable_nonce=' . wp_create_nonce( 'yoast-seo-disable-nonce' ) ), 301 );
+		exit;
+	}
+
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'yoast_seo_disable' ) ) && ! empty( $listener_plugin ) && ( $listener_nonce = filter_input( INPUT_GET, '_yoast_disable_nonce' ) ) && wp_verify_nonce( $listener_nonce, 'yoast-seo-disable-nonce' ) ) {
```

Comment:
I would put the content of this if in a function like `yoast_seo_toggle_deactive_plugin` or something like that. Just to keep it small. Just because the score in codeclimate is getting worse